src/classes/EP_TaskDependencyCheck_TDTM.cls

```diff
@@ -0,0 +1,155 @@
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2016
+* @group Engagement Plans
+* @description Trigger Handler on Engagement Plan Tasks 
+*/
+public class EP_TaskDependencyCheck_TDTM extends TDTM_Runnable {
+
+    /** @description maximum number of tasks to check in a dependency chain */
+    private static Integer maxCount = 500;
+    
+    /*******************************************************************************************************
+    * @description Trigger Handler on Engagement Plan Tasks that prevents recursion on dependent tasks.
+    * @param listNew the list of Engagement Plan Tasks from trigger new. 
+    * @param listOld the list of Engagement Plan Tasks from trigger old. 
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.). 
+    * @param objResult the describe for Engagement Plan Tasks 
+    * @return dmlWrapper.  
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> listNew, List<SObject> listOld, 
+        TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+        // A list of the tasks that require validation.
+        List<Engagement_Plan_Task__c> listToValidate = new List<Engagement_Plan_Task__c>();
+        System.debug('List to validate: ' + listToValidate);
+        // AFTER INSERT
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
+            for (SObject newSO : listNew) {
+                Engagement_Plan_Task__c newEPTask = (Engagement_Plan_Task__c)newSO;
+                if (newEPTask.Dependent_Task__c != null) {
+                    listToValidate.add(newEPTask);
+                }
+            }
+            if (listToValidate.size() > 0) {
+                validateTasks(listToValidate);
+                System.debug('List to validate: ' + listToValidate);
+            }
+        }
+
+        // AFTER UPDATE
+        if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+            Map<Id, Engagement_Plan_Task__c> mapOld = new Map<Id, Engagement_Plan_Task__c>((List<Engagement_Plan_Task__c>)listOld);
+
+            for (SObject newSO : listNew) {
+                Engagement_Plan_Task__c newEPTask = (Engagement_Plan_Task__c)newSO;
+                Engagement_Plan_Task__c oldEPTask = mapOld.get(newEPTask.Id);
+
+                if ((newEPTask.Dependent_Task__c != null
+                    && newEPTask.Dependent_Task__c != oldEPTask.Dependent_Task__c)
+                        || newEPTask.Engagement_Plan_Template__c != oldEPTask.Engagement_Plan_Template__c) {
+                        listToValidate.add(newEPTask);
+                }
+            }
+            if (listToValidate.size() > 0) {
+                validateTasks(listToValidate);
+                System.debug('List to validate: ' + listToValidate);
```

Comment:
In cases when you actually want a debug statement in production code, use the UTIL_Debug class instead.